doc/arkode/guide/source/Usage/User_callable.rst

```diff
@@ -1824,6 +1825,49 @@ Specify if the implicit RHS is deduced after a nonlinear solve  :c:func:`ARKodeS
    .. versionadded:: x.y.z
 
 
+.. c:function:: int ARKodeSetAutonomous(void* arkode_mem, sunbooleantype autonomous)
+
+   Specifies that the implicit portion of the problem is autonomous i.e., does
+   not explicitly depend on time.
+
+   When using an implicit or ImEx method with the trivial predictor, this option
+   enables reusing the implicit right-hand side evaluation at the predicted
+   state across stage solves within a step. This reuse reduces the total number
+   of implicit RHS function evaluations.
+
+   :param arkode_mem: pointer to the ARKODE memory block.
+   :param autonomous: flag denoting if the implicit RHS function,
+                      :math:`f^I(t,y)`, is autonomous (``SUNTRUE``) or
+                      non-autonomous (``SUNFALSE``).
+
+   :retval ARK_SUCCESS: the function exited successfully.
+   :retval ARK_MEM_NULL: ``arkode_mem`` was ``NULL``.
+   :retval ARK_ILL_INPUT: an argument had an illegal value.
+   :retval ARK_STEPPER_UNSUPPORTED: implicit solvers are not supported by the
+                                    current time-stepping module.
+
+   .. warning::
+
+      Results may differ when combining :c:func:`ARKodeSetAutonomous` with
+      :c:func:`ARKodeSetDeduceImplicitRhs` and using a stiffly accurate method
+      with the trivial predictor as the deduced implicit right-hand side (RHS)
+      value will be reused in the initial residual computation rather than
+      evaluating the implicit RHS function e.g.,
+      ``examples/arkode/C_serial/ark_brusselator.c``.
+
+      Similarly programs that save computations within the RHS function for
+      reuse elsewhere will need to be updated to account for the RHS function
+      reuse with :c:func:`ARKodeSetAutonomous`. For example,
+      ``examples/arkode/C_serial/ark_KrylovDemo_prec.c`` reuses rates computed
+      in the RHS function in the preconditioner setup. If
+      :c:func:`ARKodeSetAutonomous` was enabled in this example, the rates saved
+      initially would be from the RHS function evaluations used to determine the
+      initial step size rather than the RHS evaluation in the nonlinear residual
+      with the predictor.
```

Comment:
I anticipate that this statement will just confuse users.  How are they to know that they are reusing too-old of RHS values in their routines, and is this really a problem?  Some clear guidance on this is needed.  

For example, "Care must be taken in user-supplied functions that utilize stored values of the implicit RHS function (:c:type:`ARKLsJacFn`, :c:func:`ARKLsLinSysFn`, :c:func:`ARKLsJacTimesVecFn`, :c:func:`ARKLsJacTimesSetupFn`, :c:func:`ARKLsPrecSetupFn`, :c:func:`ARKLsPrecSolveFn`).  In these functions, the input vector of RHS function evaluations may come from a significantly earlier phase of the simulation than the current implicit solve (e.g., determination of the initial step size).  For problems where this may be problematic, we recommend that the user-supplied function re-evaluate the implicit RHS function internally when using :c:func:`ARKodeSetAutonomous`."

Reply:
> input vector of RHS function evaluations may come from a significantly earlier phase of the simulation than the current implicit solve

The RHS input to the functions listed will still be consistent with the state input to these functions.

The problem in the KrylovDemo example is that the "user" assumes that the nonlinear residual will be called and will evaluate the RHS before the preconditioner setup is called. Based on this assumption, the user's RHS functions saves the rates computed in the RHS and reuses them in the preconditioner setup. Enabling `SetAutonomous` breaks this assumption. For this example, the solution is the user should not save these values but instead evaluate the rates with the state input to the preconditioner setup.